cpp/src/io/orc/stripe_enc.cu

```diff
@@ -618,6 +617,100 @@ inline __device__ void lengths_to_positions(volatile T* vals, uint32_t numvals,
 static const __device__ __constant__ int32_t kTimeScale[10] = {
   1000000000, 100000000, 10000000, 1000000, 100000, 10000, 1000, 100, 10, 1};
 
+template <int block_size, typename Storage>
+static __device__ void encode_null_mask(orcenc_state_s* s,
+                                        bitmask_type const* pushdown_mask,
+                                        Storage& scan_storage,
+                                        int t)
+{
+  if (s->stream.ids[CI_PRESENT] < 0) return;
```

Comment:
not sure if I understand the question, hopefully this answers it:
If the current (child) column does not have a null mask, we don't need to encode the present stream even if the parent has a pushdown mask. 
If the current (child) column also has children, we don't need to do anything here since the pushdown masks are recomputed before the encode and the grandparent pushdown mask will be reused and passed down.